formats/cbor/commonTest/src/kotlinx/serialization/cbor/CborReaderTest.kt

```diff
@@ -104,4 +108,338 @@ class CborReaderTest {
             )
         }
     }
+
+    @Test
+    fun testIgnoreUnknownKeysFailsWhenKotlinClassIsMissingKeys() {
+        // with maps & lists of indefinite length
+        assertEquals(
+            expected = "Field 'a' is required, but it was missing",
```

Comment:
Asserting against the `Exception`'s message text which is fragile (if `SerializationException` message language changes it fails this unrelated test).

Is simply asserting against the exception type sufficient? In other words, should I remove the outer `assertEquals` and just keep the `assertFailsWith`?

Reply:
We have `assertFailsWithMessage` utility function in json tests — I think you can copy paste it to cbor test utils, too. We usually find it sufficient because exception texts rarely change